### PR TITLE
Update Contributing guide to reflect new processes

### PR DESCRIPTION
**Pull request by vaidehijoshi**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [x] Documentation Update

## Description

A section of our contributing guide needs some updating to reflect our new internal RFC process and how we use forem.dev. This seemed like a quick fix, so I just whipped up a PR for it 😸 

## Related Tickets & Documents

N/A, but something that needed doing!

## QA Instructions, Screenshots, Recordings

Check for typos + formatting 😂 

### UI accessibility concerns?

A docs update, so hopefully not! 🙃 

## Added tests?

- [ ] Yes
- [x] No, and this is why: _this a docs update my buddy_
- [ ] I need help with writing tests

## Added to documentation?

- [x] [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] README
- [ ] No documentation needed

### CODE REVIEW

File: docs/contributing/forem.md [hunk 1]
```diff
@@ -63,11 +68,21 @@ issue exists, please claim that issue and link it in your PR, otherwise creating
...
+team uses an internal RFC ("request for comments") process to assess and
+prioritize new features. This process is intended to provide a consistent and
+standardized path for new changes to enter the Forem ecosystem. If you'd like to
+propose a new feature, please visit [forem.dev](https://forem.dev) to start a
+discussion around a new feature (or chime in on a pre-existing discussion!).
+
+There may be some open issues in our repository that we think evolve into
+impactful features. For such issues, we use the
+[potential RFC](https://github.com/forem/forem/labels/potential%20RFC) label in
+order to highlight the potential feature to the Forem core team member so that
```
Comment (juliannatetreault) [inline docs/contributing/forem.md:81]:
```suggestion
order to highlight the potential feature to the Forem core team members so that
```

**Review by juliannatetreault — APPROVED** [review]
This looks great! ✨

**Review by joshpuetz — APPROVED** [review]

**Review by katiedavis — APPROVED** [review]
🏅

File: docs/contributing/forem.md (hunk 1 above)
Comment (vaidehijoshi) [reply docs/contributing/forem.md:81]: AHHHH THANK YOU 🙃

**Pull request merged**